### ISSUE TITLE
Breaking changes in CloudFormation schema for ApiGatewayV2 Integration

### ISSUE THREAD

**Issue by flostadler** [pulumi/pulumi-aws-native#1680]
The CloudFormation schema downloaded from https://schema.cloudformation.us-east-1.amazonaws.com/CloudformationSchema.zip started to have substantial changes for the  `AWS::ApiGatewayV2::Integration` resource since August 23rd. None of these changes are represented in the CloudFormation docs here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-integration.html#cfn-apigatewayv2-integration-responseparameters. 

In detail:
- The `ResponseParameter` and `ResponseParameterList` definitions are no longer present. Instead the `ResponseParameters` property is simply of type `object` now.
- The `RequestParameters` and `RequestTemplates` properties also lost their `patternProperties` and are simply of type `object` now as well.
- The `IntegrationId` property has been renamed to `Id`.
- The `ApiId` property is no longer a primary identifier

This would be a breaking change for users of aws-native. We opened an issue with AWS to see if those changes are expected: https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/2114.

**Comment by flostadler** [comment]
This appears to have been a rollback. It was introduced here ~1w ago https://github.com/pulumi/pulumi-aws-native/commit/ca0520474b26b492f7fd6c2509f28ac458b548f1.

What's weird though is that it was marked as supported ~2 months ago: [https://github.com/pulumi/pulumi-aws-native/commit/bd6319a165e95a4fb3606d0381fe4cc35009c1aa#diff-1ac835cc58d7899e9299c[…]f1bd53fe25fd4189b72e168eR40](https://github.com/pulumi/pulumi-aws-native/commit/bd6319a165e95a4fb3606d0381fe4cc35009c1aa#diff-1ac835cc58d7899e9299c7570151c7b0d7732c78f1bd53fe25fd4189b72e168eR40)
But when trying to use it you got:
```
An error occurred (UnsupportedActionException) when calling the ListResources operation: The resource AWS::ApiGatewayV2::Integration is not yet supported via Cloud Control API
```

**Comment by flostadler** [comment]
Update from AWS: https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/2114#issuecomment-2307613166

tl;dr
They rolled it back because of the breaking change, they'll roll it out again in the next 1-2 weeks. But this time with backwards compatibility. Given that the resource wasn't really usable I don't expect this to have any effects on pulumi users.

**Comment by flostadler** [comment]
Problems like this could be prevent by having some sort of type freezing mechanism: https://github.com/pulumi/pulumi-aws-native/issues/1684
